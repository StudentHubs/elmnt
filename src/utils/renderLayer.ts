import { nest } from 'recompose';

import { Comp } from '../typings';

export default function renderLayer(LayerComponent: Comp<any>) {
  return (InnerComponent: Comp<any>) => nest(LayerComponent, InnerComponent);
}
