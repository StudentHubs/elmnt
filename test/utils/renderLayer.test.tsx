import * as React from 'react';
import * as renderer from 'react-test-renderer';

import renderLayer from '../../src/utils/renderLayer';

describe('utils: renderLayer', () => {

  it('basic', () => {

    const Comp = renderLayer(({ style, children }) =>
      <div style={style}>{children}</div>
    )(({ style }) =>
      <p style={style}>Hello World!</p>
    );

    expect(renderer.create(
      <Comp style={{ color: 'red' }} />
    ).toJSON()).toMatchSnapshot();

  });

});
