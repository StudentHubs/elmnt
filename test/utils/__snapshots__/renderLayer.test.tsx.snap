// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils: renderLayer basic 1`] = `
<div
  style={
    Object {
      "color": "red",
    }
  }
>
  <p
    style={
      Object {
        "color": "red",
      }
    }
  >
    Hello World!
  </p>
</div>
`;
